clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -1228,6 +1270,17 @@ private int partition(ProducerRecord<K, V> record, byte[] serializedKey, byte[]
                         record.topic(), record.key(), serializedKey, record.value(), serializedValue, cluster);
     }
 
+    private void throwIfInvalidGroupMetadata(ConsumerGroupMetadata groupMetadata) {
+        if (groupMetadata == null) {
+            throw new IllegalStateException("Consumer group metadata could not be null");
+        } else if (groupMetadata.groupId() == null) {
```

Comment:
nit: we may as well move this check into `ConsumerGroupMetadata` since we have some other null checks there.

Reply:
The check is a little bit over-specific for ConsumerGroupMetadata itself, which is why I put advanced check in here so that people could construct group metadata in error format as they want.

Reply:
Can you elaborate why this is different from e.g. the memberId?